### ISSUE TITLE
NVIDIA cards cause 1-3s scene preview delay

### ISSUE THREAD

**Issue by Dillybob1992** [godotengine/godot#14850]
**Godot version:**
3.0, December 18th's latest compiled

**OS/device including version:**
Windows 7 64-bit

**Issue description:**
After a ton of trial and error, this issue has been finally isolated.  
When scene previewing a game on a GTX 950 (or mostly all NVIDIA cards) there is a small 1-3 second delay.  This is weird because when running a exported game, the delay isn't there. Initially, OpenGL initialization was thought to be the issue. However, that is not the case.

You can replicate this by:

- Load up Godot Editor.exe
- Load up Godot Editor.exe again (same file) **and the delay will appear**.  

That same delay is what happens during scene previewing as well (example cmd that gets run from editor):

```
"C:\Users\Test\Desktop\GodotEditor.exe" "--path" "C:/Users/Test/Documents/1234" "--remote-debug" "127.0.0.1:49153" "--allow_focus_steal_pid" "3092" "--position" "448,
```

This issue cannot be replicated on any AMD card. 

Possible temporary solution:
- Ability to choose your own executable to be run in the Godot Editor. _Loading up a different Godot executable doesn't cause the delay_

**Comment by groud** [comment]
A delay between what and what ? I don't understand exactly the problem.

**Comment by Zireael07** [comment]
I believe the issue has been reported elsewhere, there seems to be a brief delay between launching the scene/game window and it actually rendering (a flash of white IIRC). Can't find the other issue atm.

**Comment by Dillybob1992** [comment]
@Zireael07 Yeah, that's what was originally reported and it was thought to be an openGL initialization lag (in the other issue), however, unfortunately I don't believe that was the core issue after finding out how to replicate this. 

@groud Sorry, will add a gif to show. 

## Instant load
![](https://i.gyazo.com/9c973a118cba5856c20453e22eea5abd.gif)

## When an executable Godot Editor.exe is already loaded:
![](https://i.gyazo.com/0ffe95e3eab6057436cea1b02694cf66.gif)


In fact, the finding are even more odd. The more opened instances you have the editor, it scales linearly to the amount of lag the next one takes to initialize.

This is not replicable on an AMD card

**Comment by Dillybob1992** [comment]
Okay, I created a temporary fix for this and loving the GTX 950 instant previews!

This has to be an internal driver issue that is **haunting Windows user**s, and is unfortunately out of the hands of @reduz and other core developers.

Line 162, in `editor_run.cpp`:
`String exec = OS::get_singleton()->get_executable_path();`

Replaced with:
`String exec = "C:\\Builds\\SEPARATE_EXECUTABLE.exe";`

After, **duplicate** your newly compiled executable and rename it to:
`PATH_TO_A_GODOT_SEPARATE_EXECUTABLE`

Now, your scene previews will be instant on GTX 950s / NVIDIA cards:

![](https://i.gyazo.com/db844da7f6a9043201c1676c233a98c0.gif)

Compared to:

![](https://i.gyazo.com/a71304102584688e2489e1fba4c74420.gif)

If possible, maybe a custom, entirely optionable executable path could be added to the Editor Settings. So the user doesn't have to hard core anything and re-compile :)


### Disclaimer
I'm not a C++ developer so I have no idea if what I did breaks anything, but I have not noticed any issues at all. It helps a lot though, and by no means should it be used in production, etc. I'm just trying to help with a temporary fix for dumb NVIDIA driver issues.

**Comment by Dillybob1992** [comment]
Posted a report on the official NVIDIA forums [here](https://forums.geforce.com/default/topic/1032562/geforce-drivers/opengl-initialization-delay-scales-linearly-with-the-amount-of-opened-instances/), hopefully someone from NVIDIA can help reduz and other Godot core developers with finding a solution!
